bank_classes/account_class.rb

```diff
@@ -15,11 +16,11 @@ class Account
   attr_reader :balance
 
 # - A new account should be created with an ID and an initial balance
-    def initialize
+    def initialize(open_deposit_min = 0)
       array_of_accounts = CSV.read("./support/accounts.csv")
       @account_num = array_of_accounts[@@number_of_accounts][0]
       initial_deposit = array_of_accounts[@@number_of_accounts][1]
-      @balance = open_deposit_check(initial_deposit.to_i)
+      @balance = open_deposit_check(initial_deposit.to_i, open_deposit_min)
       @open_date = array_of_accounts[@@number_of_accounts][2]
       @@number_of_accounts += 1
     end
```

Comment:
In your `self.all` method below it might make more sense to use an `each` loop on the `array_of_accounts` rather that using a while loop